src/frontend/admin/src/components/presentational/table/useTableResourcePagination.ts

```diff
@@ -0,0 +1,53 @@
+import { useState } from "react";
+import { GridPaginationModel } from "@mui/x-data-grid";
+import { useDebouncedCallback } from "use-debounce";
+import { Resource, useResources } from "@/hooks/useResources";
+import { Maybe } from "@/types/utils";
+import { DEFAULT_PER_PAGE_SIZE } from "@/utils/constants";
+
+interface Props<T extends Resource> {
+  initialItemsPerPage?: number;
+  initialPage?: number;
+  useResource: ReturnType<typeof useResources<T>>;
+}
+
+export const useTableResourcePagination = <T extends Resource>({
+  initialItemsPerPage = DEFAULT_PER_PAGE_SIZE,
+  initialPage = 0,
+  useResource,
+}: Props<T>) => {
+  const [query, setQuery] = useState<Maybe<string>>();
+  const [currentPage, setCurrentPage] = useState(initialPage);
+  const [pageSize, setPageSize] = useState(initialItemsPerPage);
+
+  const resource = useResource(
+    {
+      query,
+      page: currentPage + 1,
+    },
+    { keepPreviousData: true },
+  );
+
+  const debouncedSetQuery = useDebouncedCallback((term: string) => {
+    setQuery(term);
+    setCurrentPage(0);
+  }, 300);
```

Comment:
A global constant for debounce delay could be useful if we want a consistent behavior across the app? WDYT?